Validation/RecoB/python/BDHadronTrackValidation_cff.py

```diff
@@ -13,15 +13,15 @@
 from SimTracker.TrackerHitAssociation.tpClusterProducer_cfi import *
 BDHadronTrackMonitoringAnalyze.PatJetSource = cms.InputTag('patJetsBDHadron')
 
-bdHadronTrackValidationSeq = cms.Sequence(patJetCorrections
-                                        * patJetCharge
-                                        * patJetPartonMatch
-                                        * patJetGenJetMatch
-                                        * patJetFlavourIdLegacy
-                                        * patJetFlavourId
-                                        * patJetsBDHadron
-                                        * tpClusterProducer
-                                        * BDHadronTrackMonitoringAnalyze
+bdHadronTrackValidationSeq = cms.Sequence(BDHadronTrackMonitoringAnalyze,
+                                          cms.Task(patJetCorrectionsTask
+                                                   , patJetCharge
```

Comment:
hi @Dr15Jones - I'm wondering if the real fix is to remove all of these pat* modules from the sequence (in line with them running inside the PAT sequence already)

Reply:
That I can't answer. I was trying for the minimum changes possible. From the framework end, having a Path depend on products from another Path is good if we want to be able to be able to be able to break those Paths into different Processes at some time in the future. Else, the framework would much prefer a Path be fully self contained where all EDProducers are on Tasks and not Sequences.

Reply:
hi @Dr15Jones - the idea of self contained paths would be a migration.. everything is built up from the opposite approach (sequential paths that rely on each other),

unfortunately I see that this module is monitoring a module that no one has asked to run in miniAOD (patJetsBDHadron at least). This means that potentially we change miniAOD results with or without the validation sequence...I'm not sure I know the motivation for that yet.

@slava77 @arizzi

Reply:
The problem we have here is the existing code is an amalgam of both 'semi-self contained' and 'sequences depending on other sequences'. There is one sequence which wants some earlier sequence to define what it reads while a later sequence in the same path tries to be 'self contained' and has the modules needed by the earlier sequence.

Now keep in mind my change doesn't do anything to the existing job results since all these `pat*` modules exist on a `Path` which is guaranteed to run before the `EndPath`.

Reply:
@davidlange6 on the miniAOD side, are you worried that some existing module used in the miniAOD asks for one of these `pat*` and right now since it isn't there they silently fail but with this change they would now succeed?

Reply:
Right - that or a missing/added data product in the output.